zkevm-circuits/src/evm_circuit/execution/error_oog_log.rs

```diff
@@ -0,0 +1,366 @@
+    },
+    table::CallContextFieldTag,
+    util::Expr,
+};
+use eth_types::Field;
+use eth_types::{evm_types::GasCost, evm_types::OpcodeId};
+use halo2_proofs::{circuit::Value, plonk::Error};
+
+#[derive(Clone, Debug)]
+pub(crate) struct ErrorOOGLogGadget<F> {
+    opcode: Cell<F>,
+    // memory address
+    memory_address: MemoryAddressGadget<F>,
+    // constrain gas left is less than gas cost
+    memory_expansion: MemoryExpansionGadget<F, 1, N_BYTES_MEMORY_WORD_SIZE>,
+    insufficient_gas: LtGadget<F, N_BYTES_GAS>,
+    rw_counter_end_of_reversion: Cell<F>,
+    restore_context: RestoreContextGadget<F>,
+}
+
+impl<F: Field> ExecutionGadget<F> for ErrorOOGLogGadget<F> {
+    const NAME: &'static str = "ErrorOutOfGasLOG";
+
+    const EXECUTION_STATE: ExecutionState = ExecutionState::ErrorOutOfGasLOG;
+
+    fn configure(cb: &mut ConstraintBuilder<F>) -> Self {
+        let opcode = cb.query_cell();
+        // constrain op code is Log*
+        cb.require_in_set(
+            "ErrorOutOfGasLOG happens in Log*",
+            opcode.expr(),
+            vec![
+                OpcodeId::LOG0.expr(),
+                OpcodeId::LOG1.expr(),
+                OpcodeId::LOG2.expr(),
+                OpcodeId::LOG3.expr(),
+                OpcodeId::LOG4.expr(),
+            ],
+        );
+        let mstart = cb.query_cell_phase2();
+        let msize = cb.query_word_rlc();
+        let rw_counter_end_of_reversion = cb.query_cell();
+
+        // Pop mstart_address, msize from stack
+        cb.stack_pop(mstart.expr());
+        cb.stack_pop(msize.expr());
+
+        // Note: no need to check not in static call, since write protection error will
+        // handle it.
+        let topic_count = opcode.expr() - OpcodeId::LOG0.as_u8().expr();
+
+        // check memory
+        let memory_address = MemoryAddressGadget::construct(cb, mstart, msize);
+
+        // Calculate the next memory size and the gas cost for this memory
+        // access
+        let memory_expansion = MemoryExpansionGadget::construct(cb, [memory_address.address()]);
+
+        let gas_cost = GasCost::LOG.as_u64().expr()
+            + GasCost::LOG.as_u64().expr() * topic_count
+            + 8.expr() * memory_address.length()
+            + memory_expansion.gas_cost();
+
+        // Check if the amount of gas available is less than the amount of gas
+        // required
+        let insufficient_gas = LtGadget::construct(cb, cb.curr.state.gas_left.expr(), gas_cost);
+        cb.require_equal(
+            "log* gas left is less than gas required ",
+            insufficient_gas.expr(),
+            1.expr(),
+        );
+
+        // current call must be failed.
+        cb.call_context_lookup(false.expr(), None, CallContextFieldTag::IsSuccess, 0.expr());
+
+        cb.call_context_lookup(
+            false.expr(),
+            None,
+            CallContextFieldTag::RwCounterEndOfReversion,
+            rw_counter_end_of_reversion.expr(),
```

Comment:
The `rw_counter_end_of_reversion` value is used in a read lookup, but there's no constraint associated to it.  Why is it read then?  I see that other errors do this:
```
rw_counter_end_of_step = cb.curr.state.rw_counter.expr() + cb.rw_counter_offset() - 1
require rw_counter_end_of_reversion == rw_counter_end_of_step + reversible_counter
```

Reply:
Agree. I thought it had to do with checking if the reversion was happening for the entire TX and so we were at the last RW, but this is done below on a different fashion.

Reply:
it is missing , added in [920c06b](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1116/commits/920c06b0a2e727c0fd3a3c38db96eed0530b2c01) as others